contracts/lib.rs

```diff
@@ -998,17 +1076,15 @@ pub mod prosopo {
                 .captcha_solution_commitments
                 .get(&captcha_solution_commitment_id)
                 .ok_or_else(err_fn!(Error::CaptchaSolutionCommitmentDoesNotExist))?;
-            let mut user = self.dapp_users.get(&commitment.account).ok_or_else(err_fn!(Error::DappDoesNotExist))?;
+            self.dapp_users.get(&commitment.account).ok_or_else(err_fn!(Error::DappUserDoesNotExist))?;
 
             // only make changes if commitment is Pending approval or disapproval
             if commitment_mut.status == CaptchaStatus::Pending {
                 commitment_mut.status = CaptchaStatus::Approved;
-                user.correct_captchas += 1;
-                user.last_correct_captcha = commitment.completed_at;
-                user.last_correct_captcha_dapp_id = commitment.contract;
+                self.record_captcha_result(commitment.account, commitment_mut);
+
                 self.captcha_solution_commitments
```

Comment:
Wasn't part of this issue to avoid storing captcha solution commitments indefinitely? If so, we'd need to remove old ones from this storage item: `self.captcha_solution_commitments`.

Reply:
It looks like the only place they're removed from is the `user.history`

Reply:
have addressed this in the latest commits